test/fixtures/test-runner/output/test-timeout-flag.js

```diff
@@ -0,0 +1,25 @@
+// Flags: --test-timeout=20
+'use strict';
+const { describe, test, afterEach } = require('node:test');
+const { setTimeout } = require('node:timers/promises');
+
+let timeout;
+
+describe('--test-timeout is set to 20ms', () => {
+  test('should timeout after 20ms', async () => {
+    timeout = await setTimeout(2000);
```

Comment:
I would make the timeout here and in the next test even larger. There is no harm in making it really large because it's going to be unref'ed anyway.